docs/content/config/advanced/ipv6.md

```diff
@@ -2,43 +2,49 @@
-+    volumes:
-+      - /var/run/docker.sock:/var/run/docker.sock:ro
-+      - /lib/modules:/lib/modules:ro
-
-@@ ... @@ networks:
-
-+  default:
-+    driver: bridge
-+    enable_ipv6: true
-+    ipam:
-+      driver: default
-+      config:
-+        - subnet: fd00:0123:4567::/48
-+          gateway: fd00:0123:4567::1
-```
+!!! bug "Ample Opportunities for Issues"
+
+    Numerous bug reports have been raised in the past about IPv6. Please make sure your setup around DMS is correct when using IPv6!
+
+## Issues with Docker
+
+### Background
 
-## Further Discussion
+If your host supports IPv6, then DMS can automatically accept IPv6 connections. The issue is with _Docker_'s [NAT64]:  incoming mails will fail SPF checks because they will appear to come from the IPv4 gateway ((most likely `172.20.0.1`)) that Docker is using to proxy the IPv6 connection.
 
-See [#1438][github-issue-1438]
+To read on, issues [#1438][github-issue-1438] & [#3057][github-issue-3057] provide material for further discussion.
 
+[wikipedia-nat64]: https://en.wikipedia.org/wiki/NAT64
 [github-issue-1438]: https://github.com/docker-mailserver/docker-mailserver/issues/1438
+[github-issue-3057]: https://github.com/docker-mailserver/docker-mailserver/pull/3057#issuecomment-1416706615
+
+### Solution
+
+The issue can be solved by supporting IPv6 connections all the way to the DMS container.
+
+You definitely want to make sure Docker has IPv6 enabled. The [official Docker documentation on enabling IPv6][docker-docs-enable-ipv6] provides you information on how to do that. Thereafter, if you want to use container networking with IPv6, make sure you have the following in `/etc/docker/daemon.json`:
+
+```json
+{
+  "ip6tables": true,
+  "experimental" : true,
+  "userland-proxy": true
+}
+```
+
+You'll need to restart the daemon if it's running, not just reload it. The above enables the IPv6 NAT which will avoid the routing to IPv4, so long as the container is on an ipv6 network that we'll configure next. `experimental` is required currently for `ip6tables` to work; we think `userland-proxy` is too (_although this setting should be enabled by default, there is upstream talk to switch to disabled by default though_).
+
+Then you need to configure a network for your container, in `docker-compose.yaml` the default bridge network isn't the same as the default bridge config in `/etc/docker/daemon.json`, that only applies to `docker` CLI (eg `docker run`). Instead you can [override the default network like I detailed here](https://github.com/nginx-proxy/nginx-proxy/issues/133#issuecomment-1368745843):
+
+```yaml
+networks:
+  # Overrides the `default` compose generated network, avoids needing to attach to each service:
+  default:
+    enable_ipv6: true
+    # An IPv4 subnet is implicitly configured, IPv6 needs to be specified:
+    ipam:
+      config:
+        - subnet: fd00:cafe:babe::/48
+```
+
+[docker-docs-enable-ipv6]: https://docs.docker.com/config/daemon/ipv6/
```

Comment:
````suggestion
Next, configure a network for your container with any of these:
- [User-defined networks via `docker network create` or `compose.yaml`][docker-docs-ipv6-create-custom]
- [Default docker bridge][docker-docs-ipv6-create-default] (_docker CLI only, not `compose.yaml`_)
- [Default network for a `compose.yaml`][ipv6-config-example] (_ `/etc/docker/daemon.json` settings for default bridge do not apply, instead override the generated `default` network_)

If you've [configured IPv6 address pools in `/etc/docker/daemon.json`][docker-docs-ipv6-supernets], you do not need to specify a subnet explicitly. Otherwise if you're unsure what value to provide, here's a quick guide (_Tip: Prefer IPv6 ULA, it's the least hassle_):
- `2001:db8` and similar looking addresses are [reserved for documentation][wikipedia-ipv6-reserved]. **Don't use this** for your private subnet.
- `fd00:cafe:face:feed::/64` is an example of a [IPv6 ULA subnet][wikipedia-ipv6-ula]. ULA addresses are akin to the [private IPv4 subnets][wikipedia-ipv4-private] you may already be familiar with. You can use that example, or choose your own ULA address. This is a good choice for getting Docker containers to their have networks support IPv6 via NAT like they already do by default with IPv4.
- IPv6 without NAT, using public address space like your server is assigned belongs to an [IPv6 GUA subnet][wikipedia-ipv6-gua].
    - Typically these will be a `/64` block assigned to your host, but this varies by provider.
    - These addresses do not need to publish ports of a container to another IP to be publicly reached, you will want a firewall configured to manage which ports are accessible instead as no NAT is involved. Note that this may not be desired if the container should also be reachable via the host IPv4 public address.
    - You may want to subdivide the `/64` into smaller subnets for Docker to use only portions of the `/64`. This can reduce some routing features, and [require additional setup / management via a NDP Proxy][gh-pull-3244-gua] for your public interface to know of IPv6 assignments managed by Docker and accept external traffic.

[docker-docs-enable-ipv6]: https://docs.docker.com/config/daemon/ipv6/
[docker-docs-ipv6-create-custom]: https://docs.docker.com/config/daemon/ipv6/#create-an-ipv6-network
[docker-docs-ipv6-create-default]: https://docs.docker.com/config/daemon/ipv6/#use-ipv6-for-the-default-bridge-network
[docker-docs-ipv6-supernets]: https://docs.docker.com/config/daemon/ipv6/#dynamic-ipv6-subnet-allocation

[ipv6-config-example]: https://github.com/nginx-proxy/nginx-proxy/issues/133#issuecomment-1368745843
[wikipedia-ipv6-reserved]: https://en.wikipedia.org/wiki/IPv6_address#Documentation
[wikipedia-ipv4-private]: https://en.wikipedia.org/wiki/Private_network#Private_IPv4_addresses
[wikipedia-ipv6-ula]: https://en.wikipedia.org/wiki/Unique_local_address
[wikipedia-ipv6-gua]: https://en.wikipedia.org/wiki/IPv6#Global_addressing

[gh-pull-3244-gua]: https://github.com/docker-mailserver/docker-mailserver/pull/3244#issuecomment-1528984894
````